src/object/srv_obj.c

```diff
@@ -1063,8 +1071,19 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 		D_GOTO(post, rc);
 	}
 
-	rc = obj_verify_bio_csum(rpc, iods, iod_csums, biod,
-				 cont_hdl->sch_csummer);
+	if (obj_rpc_is_update(rpc)) {
+		rc = obj_verify_bio_csum(rpc, iods, iod_csums, biod,
+					 cont_hdl->sch_csummer);
+		/** CSUM Verified on update, now corrupt to fake corruption
+		 * on disk
+		 */
+		if (DAOS_FAIL_CHECK(DAOS_CSUM_CORRUPT_DISK) && !rma) {
```

Comment:
here the data already written to disk right? just curious how would the below dcf_corrupt help?

Reply:
I was expecting that data hasn't been written here. (isn't that what bio_iod_post does?)